protocol-units/bridge/contracts/src/AtomicBridgeInitator.sol

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.22;
+
+import {IAtomicBridgeInitiator} from "./IAtomicBridgeInitiator.sol";
+import {IWETH10} from "./WETH/interfaces/IWETH10.sol";
+import {Initializable} from "@openzeppelin/contracts/proxy/utils/Initializable.sol";
+
+contract AtomicBridgeInitiator is IAtomicBridgeInitiator, Initializable {
+    struct BridgeTransfer {
+        uint256 amount;
+        address originator;
+        bytes32 recipient;
+        bytes32 hashLock;
+        uint256 timeLock;
+        bool completed;
+    }
+
+    mapping(bytes32 => BridgeTransfer) public bridgeTransfers;
+    IWETH10 public weth;
+    address public authority;
+
+    modifier onlyAuthorized() {
+        if (msg.sender != authority) revert Unauthorized();
+        _;
+    }
+
+    function initialize(address _weth) public initializer {
+        if (_weth == address(0)) {
+            revert ZeroAddress();
+        }
+        authority = msg.sender;
+        weth = IWETH10(_weth);
+    }
+
+    function initiateBridgeTransfer(uint256 _wethAmount, bytes32 _recipient, bytes32 _hashLock, uint256 _timeLock)
+        external
+        payable
+        returns (bytes32 _bridgeTransferId)
+    {
+        address originator = msg.sender;
+        uint256 ethAmount = msg.value;
+        uint256 totalAmount = _wethAmount + ethAmount;
+        _bridgeTransferId = keccak256(abi.encodePacked(originator, _recipient, _hashLock, _timeLock, block.timestamp));
+        // Check if the bridge transfer already exists
+        if (bridgeTransfers[_bridgeTransferId].amount != 0) revert BridgeTransferExists();
+        // Ensure there is a valid total amount
+        if (totalAmount == 0) {
+            revert ZeroAmount();
+        }
+        // If msg.value is greater than 0, convert ETH to WETH
+        if (ethAmount > 0) weth.deposit{value: ethAmount}();
```

Comment:
Are gas fees included in the ETH amount? Just to double-check, if not, it could lead to extra costs otherwise.

Reply:
I checked on this. Gas fees are deducted from `msg.value` before the function call happens. So `ethAmount` would not include gas fees.